test/schemas/tests-schema-0.0.1.json

```diff
@@ -0,0 +1,445 @@
+              "items": {
+                "required": [
+                  "locale"
+                ]
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "anyOf": [
+        {
+          "properties": {
+            "defaultTestProperties": {
+              "required": [
+                "src"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "required": [
+                  "src"
+                ]
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "$comment": "Only one assertion is required. It doesn't matter which type.",
+      "anyOf": [
+        {
+          "properties": {
+            "defaultTestProperties": {
+              "$ref": "#/$defs/anyExp"
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "$ref": "#/$defs/anyExp"
+              }
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "$defs": {
+    "defaultTestProperties": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "testType": {
+          "$ref": "#/$defs/testType"
+        },
+        "locale": {
+          "$ref": "#/$defs/locale"
+        },
+        "src": {
+          "$ref": "#/$defs/src"
+        },
+        "params": {
+          "$ref": "#/$defs/params"
+        },
+        "exp": {
+          "$ref": "#/$defs/exp"
+        },
+        "expCleanSrc": {
+          "$ref": "#/$defs/expCleanSrc"
+        },
+        "expParts": {
+          "$ref": "#/$defs/expParts"
+        },
+        "expErrors": {
+          "$ref": "#/$defs/expErrors"
+        }
+      }
+    },
+    "test": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Information about the test."
+        },
+        "testType": {
+          "$ref": "#/$defs/testType"
+        },
+        "locale": {
+          "$ref": "#/$defs/locale"
+        },
+        "src": {
+          "$ref": "#/$defs/src"
+        },
+        "params": {
+          "$ref": "#/$defs/params"
+        },
+        "exp": {
+          "$ref": "#/$defs/exp"
+        },
+        "expCleanSrc": {
+          "$ref": "#/$defs/expCleanSrc"
+        },
+        "expParts": {
+          "$ref": "#/$defs/expParts"
+        },
+        "expErrors": {
+          "$ref": "#/$defs/expErrors"
+        },
+        "only": {
+          "type": "boolean",
+          "description": "Normally not set. A flag to use during development to only run one or more specific tests."
+        }
+      }
+    },
+    "testType": {
+      "description": "The category of test.",
+      "enum": [
+        "syntax",
+        "function"
+      ]
+    },
```

Comment:
What's this used for?

Reply:
This comment and the ones @eemeli makes above suggest that we'd be well-served by a design document.

Reply:
@eemeli This was a way to indicate that different options/args are passed to the test executor for different types of test, solving a similar problem to https://github.com/unicode-org/message-format-wg/pull/769, where we need special behaviour for function tests.

For example, a value of `"testType": "function"` could automatically mock out instances of `:number` in the pattern to achieve similar behaviour to using `:test:number`. However, if #769 lands, I don't think there is a need for `testType`. I'll remove it now on that basis.

Reply:
@aphillips No problem - I'm planning to update https://github.com/unicode-org/message-format-wg/blob/main/exploration/data-driven-tests.md to document the outcomes from this PR when we reach consensus, if that works?